packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -1376,6 +1376,27 @@ const tests = {
         }
       `,
     },
+    {
+      code: normalizeIndent`
+        function Example({ sortBy }) {
+          const foo = () => useMemo(() => console.log(sortBy), [sortBy])
```

Comment:
The thing that's odd about these examples is that they break the Rules of Hooks. Concretely, here we have a Hook call inside a callback. But Hook calls should only happen at the top level.

Reply:
Yes, it's totally right. So there is no bug but only a wrong hook usage.

Reply:
Given that, I think I could close this PR because it eventually fixes a situation caused by a wrong Hook usage.
Since Hook calls should only happen at the top level, component props will be alway in `pureScopes` array.

Reply:
@gaearon sorry for coming back to this argument, I have just one question.
Is the use I made in [this](https://codesandbox.io/s/nervous-haze-0pwlf?file=/src/App.jsx) sandbox breaking the hook rule? Because I don't receive any message from `rules-of-hooks` and I am thinking it could be a legit use (even in a really contrived example) because the hook is called twice a render.

Sorry again if I am missing something.

Reply:
Totally misunderstood.
These examples are breaking the Rules of Hooks because the hook is called inside a function that is not a custom hook and neither in a JSX component.
The reason why the example in CodeSandbox is working is that the function within the hook is called begins with `use`. So...double error.

Sorry for the waste of time.